app/models/ncr_dispatcher.rb

```diff
@@ -1,67 +0,0 @@
-class NcrDispatcher < LinearDispatcher
```

Comment:
I agree that would be problematic. But what we have now was already confusing so this PR is my attempt at clarifying the logic. Ideas on a better way of dealing with the branching?